### PR TITLE
Added tests for Book and Categoies group

### PR DESCRIPTION
**Pull request by sanmartial**
Screen coverage tests:
[https://drive.google.com/file/d/1PeRSpNRzSH3GSHDJ_WpU26q91MWoUM9Y/view?usp=sharing]

### CODE REVIEW

File: src/main/java/com/globaroman/bookshopproject/controller/BookController.java [hunk 1]
```diff
@@ -65,7 +65,7 @@ public void delete(@PathVariable Long id) {
     @PreAuthorize("hasRole('ADMIN')")
     @PutMapping("/{id}")
     @ResponseStatus(HttpStatus.OK)
-    public void update(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
-        bookService.update(id, bookDto);
+    public BookDto update(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
```
Comment (sokimaaa) [inline src/main/java/com/globaroman/bookshopproject/controller/BookController.java]: validate request?

File: src/test/java/com/globaroman/bookshopproject/config/CustomMySqlContainer.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+public class CustomMySqlContainer extends MySQLContainer<CustomMySqlContainer> {
+    public static final String DB_IMAGE = "mysql:8";
+    private static CustomMySqlContainer mySqlContainer;
+
+    private CustomMySqlContainer() {
+        super(DB_IMAGE);
+    }
+
+    public static synchronized CustomMySqlContainer getInstance() {
```
Comment (sokimaaa) [inline src/test/java/com/globaroman/bookshopproject/config/CustomMySqlContainer.java:13]: bad way of implementation the Singleton pattern
synchronized is potentially bottleneck. 
```suggestion
    public static synchronized CustomMySqlContainer getInstance() {
```

File: src/test/java/com/globaroman/bookshopproject/config/CustomMySqlContainer.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    public void start() {
+        super.start();
+        System.setProperty("TEST_DB_URL", mySqlContainer.getJdbcUrl());
+        System.setProperty("TEST_DB_USERNAME", mySqlContainer.getUsername());
+        System.setProperty("TEST_DB_PASSWORD", mySqlContainer.getPassword());
+    }
+
+    @Override
+    public void stop() {
+    }
```
Comment (sokimaaa) [inline src/test/java/com/globaroman/bookshopproject/config/CustomMySqlContainer.java]: Dont stop container? may cause timeout, or memory leak issues
```suggestion
    public void stop() {
    }
```

File: src/test/java/com/globaroman/bookshopproject/controller/BookControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+                .andExpect(status().isCreated())
+                .andReturn();
+        //Then
+        BookDto actual = objectMapper.readValue(result.getResponse()
+                .getContentAsString(), BookDto.class);
+
+        Assertions.assertNotNull(actual);
+        Assertions.assertNotNull(actual.getId());
+
+        EqualsBuilder.reflectionEquals(requestDto, actual, "id");
```
Comment (sokimaaa) [inline src/test/java/com/globaroman/bookshopproject/controller/BookControllerTest.java]: you dont need ObjectMappet to read value and so on....

spring boot test already has ability to compare expect\actual results via jsonPath()
```suggestion
        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.post("/api/books")
                        .content(jsonRequest)
                        .contentType(MediaType.APPLICATION_JSON))
                .andExpect(status().isCreated())
                .andReturn();
    
```

Comment (sokimaaa) [reply src/test/java/com/globaroman/bookshopproject/controller/BookControllerTest.java]: rewrite all tests regarding this comments

File: src/test/java/com/globaroman/bookshopproject/controller/BookControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Get all books")
+    @Transactional
+    @Rollback
```
Comment (sokimaaa) [inline src/test/java/com/globaroman/bookshopproject/controller/BookControllerTest.java:83]: Any sense of using this? 
```suggestion
```

File: src/test/java/com/globaroman/bookshopproject/repository/BookRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class BookRepositoryTest {
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    private Category category;
```
Comment (sokimaaa) [inline src/test/java/com/globaroman/bookshopproject/repository/BookRepositoryTest.java]: create object in test case where you use it. 
```suggestion
```

File: src/test/java/com/globaroman/bookshopproject/config/CustomMySqlContainer.java (hunk 2 above)
Comment (sanmartial) [reply src/test/java/com/globaroman/bookshopproject/config/CustomMySqlContainer.java:13]: Probably! But just used example that showed Bogdan in the video about TestContainers

Comment (sanmartial) [reply src/test/java/com/globaroman/bookshopproject/config/CustomMySqlContainer.java:13]: Probably! But I just used example that showed Bogdan in video about TestContainers

**Pull request merged**